demonstrations/tutorial_jax_transformations.py

```
@@ -0,0 +1,207 @@
+    qml.RX(param, wires=0)
+    qml.CNOT(wires=[0, 1])
+    return qml.expval(qml.PauliZ(0))
+
+print("First run:", circuit(0.123)) # Compile overhead the first time time method is executed.
+print("Second run:", circuit(0.123)) # Much faster every time after!
+
+##############################################################################
+# Shots and Sampling with JAX.
+# ----------------------------
+# 
+# JAX was designed to have experiments be as repeatable as possible. Because of this,
+# JAX requires us to seed all randomly generated values (as you saw in the above
+# batching example). Sadly, the universe doesn't allow us to seed real quantum computers,
+# so if we want our JAX to mimic a real QC, we'll have to handle randomness ourselves.
+#
+# We recommend reading up 
+# 
+# Note: This example only applies if you are using jax.jit. Otherwise, we automatically 
+# seed and reset the random number generator for you on each call.
+#
+# To set the random number generating key, you'll have to pass the jax.random.PRNGKey
+# when constructing the device. Because of this, if you want to use jax.jit with randomness,
+# the device construction will have to happen within that jitted method.
+
+
+
+# Let's create our circuit with randomness and compile it with jax.jit.
+@jax.jit
+def circuit(key, param):
+    # Notice how the device construction now happens within the jitted method.
+    dev = qml.device("default.qubit.jax", wires=2, shots=10, prng_key=key)
+
+    # Now we can create our qnode within the circuit function.
+    @qml.qnode(dev, interface="jax")
+    def my_circuit():
+        qml.RX(param, wires=0)
+        qml.CNOT(wires=[0, 1])
+        return qml.sample(qml.PauliZ(0))
+    return my_circuit()
+
+key1 = jax.random.PRNGKey(0)
+key2 = jax.random.PRNGKey(1)
+
+# Notice that the first two runs return exactly the same results,
+# The second run has different results.
+print(f"key1: {circuit(key1, jnp.pi/2)}")

```

A warning appears when this runs in the build - we can suppress those the phrase to the `conf.py` file (there are some examples already there around line 90).

---

```/home/circleci/project/venv/lib/python3.7/site-packages/jax/_src/numpy/lax_numpy.py:5179: UserWarning: Explicitly requested dtype <class 'numpy.int64'> requested in astype is not available, and will be truncated to dtype int32. To enable more dtypes, set the jax_enable_x64 configuration option or the JAX_ENABLE_X64 shell environment variable. See https://github.com/google/jax#current-gotchas for more.
  lax._check_user_dtype_supported(dtype, "astype")```

---

If possible, better to code the example to avoid warnings. Users who run locally might encounter warning and ask us about it

---

Updated the config to run in x64 mode, silencing this error.